pkg/file/redundancy/getter/getter.go

```diff
@@ -97,10 +98,39 @@ func (g *decoder) Get(ctx context.Context, addr swarm.Address) (swarm.Chunk, err
 	}
 	select {
 	case <-g.waits[i]:
-		return swarm.NewChunk(addr, g.rsbuf[i]), nil
 	case <-ctx.Done():
-		return nil, ctx.Err()
+		select {
```

Comment:
context deadline must be respected.
move the cases inside the inner select to the outer select.

Reply:
yes, I did feel unconfortable about how this is handled. Unfortunately we cannot move from inside select to outer, but i will think about a better and correct way to handle this.

Reply:
it was not correct, fixed context logic